custom_components/plugwise/select.py

```diff
@@ -43,18 +91,31 @@ def __init__(
         self,
         coordinator: PlugwiseDataUpdateCoordinator,
         device_id: str,
+        description: PlugwiseSelectEntityDescription,
     ) -> None:
         """Initialise the selector."""
         super().__init__(coordinator, device_id)
-        self._attr_unique_id = f"{device_id}-select_schedule"
-        self._attr_name = (f"{self.device.get('name', '')} Select Schedule").lstrip()
-        self._attr_current_option = self.device.get("selected_schedule")
-        self._attr_options = self.device.get("available_schedules", [])
+        self.entity_description = description
+        self._attr_unique_id = f"{device_id}-{description.key}"
+        self._attr_name = (f"{self.device.get('name', '')} {description.name}").lstrip()
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        return self.device.get(self.entity_description.current_option)
+
+    @property
+    def options(self) -> list[str]:
+        """Return the selectable entity options."""
+        return self.device.get(self.entity_description.options, [])
 
     async def async_select_option(self, option: str) -> None:
-        """Change the selected option."""
-        await self.coordinator.api.set_schedule_state(
-            self.device.get("location"),
-            option,
-            SCHEDULE_ON,
-        )
+        """Change to the selected entity option."""
+        result = await self.async_send_api_call(option, self.entity_description.command)
```

Comment:
I'd still recommend rewriting this to the approach we did in https://github.com/home-assistant/core/blob/dev/homeassistant/components/plugwise/select.py

Reply:
I don't think that construct will work?